src/components/PopupWithForm.js

```diff
@@ -35,4 +30,12 @@ export default class PopupWithForm extends Popup {
       this._popupForm.reset();
     });
   }
+
+  saving = (saving) => {
```

Comment:
Needs correction: there is no need to create two different metjods for rendering loading, you should create just one universal.

One of the options to implement that would be adding a parameter `loadingButtonText` to the constructor of `PopupWithForm` and using it to change button text. 

```js
constructor(popupSelector, { handleSubmit, loadingButtonText }) { 
  ...
  this._submitButton = this._formElement.querySelector(...)
  this._buttonText = this._submitButton.textContent;
  this._loadingButtonText = loadingButtonText;
  …
}

showLoading() {
  this._submitButton.textContent = this._loadingButtonText
}

hideLoading() {
  this._submitButton.textContent = this._buttonText
}
``` 

The api call should look like this:

```js
popup.showLoading()
api.
  .then(...)
  .catch(...)
  .finally(() => {
    popup.hideLoading()
```